simpleflow/swf/executor.py

```diff
@@ -524,18 +531,13 @@ def schedule_task(self, a_task, task_list=None):
             self._add_start_timer_decision('resume-after-{}'.format(a_task.id))
             raise exceptions.ExecutionBlocked()
 
-        # Ready to schedule
-        logger.debug('executor is scheduling task {} on task_list {}'.format(
-            a_task.name,
-            task_list,
-        ))
+        # logger.debug('{}: adding decision(s) {}'.format(self._workflow_id, decisions))
         self._decisions.extend(decisions)
-        self._open_activity_count += 1
 
         # Check if we won't exceed max decisions -1
         # TODO: if we had exactly MAX_DECISIONS - 1 to take, this will wake up
         # the workflow for no reason. Evaluate if we can do better.
-        if len(self._decisions) == constants.MAX_DECISIONS - 1:
+        if len(self._decisions) >= constants.MAX_DECISIONS - 1:
```

Comment:
Not currently as `decisions` is a 1-item list :wink:
But `SignalTask.schedule` could have scheduled the signal on the children and parent workflow too, who knows?